File: app/src/main/java/com/two2/pranjal/mathisfun/IndexActivity.java
package com.two2.pranjal.mathisfun;

import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.wajahatkarim3.easyflipview.EasyFlipView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class IndexActivity extends AppCompatActivity{

    @BindView(R.id.flipView)
    EasyFlipView flipView;
    @BindView(R.id.frontQuestion)
    TextView frontQuestion;
    @BindView(R.id.backAnswer)
    TextView backAnswer;
    @BindView(R.id.skipIndexCard)
    FloatingActionButton skipButton;
    @BindView(R.id.flipIndexCard)
    FloatingActionButton flipButton;
    ProblemFinder problemFinder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_index);
        ButterKnife.bind(this);
        Intent intent= getIntent();
        String type= intent.getStringExtra(getResources().getString(R.string.type));
        String difficulty= intent.getStringExtra(getResources().getString(R.string.difficulty));
        problemFinder= new ProblemFinder(type,difficulty,this);
        ProblemObject problemObject= problemFinder.getProblem();
        String question= problemObject.getQuestion1()+" "+problemObject.getType()+" "+problemObject.getQuestion2();
        frontQuestion.setText(question);
        backAnswer.setText(problemObject.getAnswer());

        skipButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                nextCard();
            }
        });

        flipButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                flipView.flipTheView();
            }
        });
    }

    private void nextCard(){
        ProblemObject problemObject= problemFinder.getProblem();
        String question= problemObject.getQuestion1()+" "+problemObject.getType()+" "+problemObject.getQuestion2();
        frontQuestion.setText(question);
        if(flipView.isBackSide()){
            flipView.flipTheView();
        }
        backAnswer.setText(problemObject.getAnswer());
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        Intent intent= new Intent(this, MainScreen.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
    }
}

File: app/src/main/java/com/two2/pranjal/mathisfun/IndexActivity.kt
package com.two2.pranjal.mathisfun
//
//import android.content.Intent
//import android.os.Bundle
//import android.widget.TextView
//import androidx.appcompat.app.AppCompatActivity
//import butterknife.BindView
//import butterknife.ButterKnife
//import com.google.android.material.floatingactionbutton.FloatingActionButton
//import com.two2.pranjal.mathisfun.MainScreen
//import com.wajahatkarim3.easyflipview.EasyFlipView
//
//class IndexActivity : AppCompatActivity() {
//    @BindView(R.id.flipView)
//    var flipView: EasyFlipView? = null
//
//    @BindView(R.id.frontQuestion)
//    var frontQuestion: TextView? = null
//
//    @BindView(R.id.backAnswer)
//    var backAnswer: TextView? = null
//
//    @BindView(R.id.skipIndexCard)
//    var skipButton: FloatingActionButton? = null
//
//    @BindView(R.id.flipIndexCard)
//    var flipButton: FloatingActionButton? = null
//    var problemFinder: ProblemFinder? = null
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_index)
//        ButterKnife.bind(this)
//        val intent = intent
//        val type = intent.getStringExtra(resources.getString(R.string.type))
//        val difficulty = intent.getStringExtra(resources.getString(R.string.difficulty))
//        problemFinder = ProblemFinder(type, difficulty, this)
//        val problemObject = problemFinder.getProblem()
//        val question = problemObject.question1 + " " + problemObject.type + " " + problemObject.question2
//        frontQuestion!!.text = question
//        backAnswer.setText(problemObject.answer)
//        skipButton!!.setOnClickListener { nextCard() }
//        flipButton!!.setOnClickListener { flipView!!.flipTheView() }
//    }
//
//    private fun nextCard() {
//        val problemObject = problemFinder.getProblem()
//        val question = problemObject.question1 + " " + problemObject.type + " " + problemObject.question2
//        frontQuestion!!.text = question
//        if (flipView!!.isBackSide) {
//            flipView!!.flipTheView()
//        }
//        backAnswer.setText(problemObject.answer)
//    }
//
//    override fun onBackPressed() {
//        super.onBackPressed()
//        val intent = Intent(this, MainScreen::class.java)
//        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
//        startActivity(intent)
//    }
//}